tests/results_benchmark.py

```diff
@@ -137,62 +171,164 @@ def __init__(self, root: Path = Path("data/raw/pipal")):
+    feature_extractor = get_feature_extractor(feature_extractor_name=feature_extractor_name, device=device)
+
+    if feature_extractor_name == 'inception':
+        distorted_images = normalize_tensor(distorted_images)
+        reference_images = normalize_tensor(reference_images)
+
+    # Create patches
+    distorted_patches = crop_patches(distorted_images, size=96, stride=32)
+    reference_patches = crop_patches(reference_images, size=96, stride=32)
+
+    # Extract features from distorted images
+    distorted_patches = distorted_patches.view(-1, *distorted_patches.shape[-3:])
+    reference_patches = reference_patches.view(-1, *reference_patches.shape[-3:])
+
+    distorted_features, reference_features = extract_features(distorted_patches, feature_extractor,
+                                                              feature_extractor_name, reference_patches)
+
+    return metric_functor(distorted_features, reference_features).cpu()
+
+
+def crop_patches(images: torch.Tensor, size=64, stride=32):
+    """Crop input images into smaller patches
+    Args:
+        images: Tensor of images with shape (batch x 3 x H x W)
+        size: size of a square patch
+        stride: Step between patches
+    """
+    patches = images.data.unfold(1, 3, 3).unfold(2, size, stride).unfold(3, size, stride)
+    patches = patches.reshape(-1, 3, size, size)
+    return patches
+
+
+def main(dataset_name: str, path: Path, metrics: List[str], device: str, feature_extractor: str) \
+        -> None:
     # Init dataset and dataloader
     dataset = DATASETS[dataset_name](root=path)
-    loader = torch.utils.data.DataLoader(dataset, batch_size=batch_size, num_workers=0)
-
-    for name in metrics:
-        gt_scores, metric_scores = eval_metric(loader, METRICS[name], device=device)
-        print(f"{name}: SRCC {abs(spearmanr(gt_scores, metric_scores)[0]):0.3f}",
+    loader = DataLoader(dataset, batch_size=1, num_workers=4)
+
+    for metric_name in metrics:
+        metric: Metric = METRICS[metric_name]
+        gt_scores, metric_scores = eval_metric(loader, metric, device=device, feature_extractor=feature_extractor)
+        print('gt_scores', gt_scores)
+        print('metric_scores', metric_scores)
+        print(f"{metric_name}: SRCC {abs(spearmanr(gt_scores, metric_scores)[0]):0.3f}",
               f"KRCC {abs(kendalltau(gt_scores, metric_scores)[0]):0.3f}")
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="Benchmark PIQ metrics")
 
-    # General
     parser.add_argument("--dataset", type=str, help="Dataset name", choices=list(DATASETS.keys()))
     parser.add_argument("--path", type=Path, help="Path to dataset")
     parser.add_argument('--metrics', nargs='+', default=[], help='Metrics to benchmark', choices=list(METRICS.keys()))
-    parser.add_argument('--batch_size', type=int, default=1, help='Batch size')
```

Comment:
Why you deleted `batch_size` option? It can be useful in case if dataset consists of same sized images.

Reply:
The script is meant to compute correlation values between MOSs and metrics' scores. If `batch_size > 1`, the script computed a single metric score for `batch_size` number of images. In this case:
1. Miscorrespondance between number of MOSs and number of metrics' scores causes a crush
2. We may try to fix it by computing some aggregates of MOSs, which I decided not to do because the usage could be error-prone

Reply:
Nice catch. I removed it when I was working with medical images of various sizes. Returning back

Reply:
Check out the issue #309